osu.Game.Rulesets.Mania/Mods/ManiaModNoLongNotes.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Mania.Objects;
+using osu.Game.Rulesets.Mods;
+using osu.Framework.Graphics.Sprites;
+using System;
+using System.Collections.Generic;
+using osu.Game.Audio;
+using osu.Game.Rulesets.Mania.Beatmaps;
+using osu.Framework.Utils;
+using osu.Game.Overlays.Settings;
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+
+
+
+namespace osu.Game.Rulesets.Mania.Mods
+{
+    public class ManiaModNoLongNotes : Mod, IApplicableAfterBeatmapConversion
+    {
+
+        public override string Name => "No Long Notes";
+
+        public override string Acronym => "NL";
+        
+        public override double ScoreMultiplier => 1;
+        
+        public override string Description => @"Turns all held notes into tap notes. No coordination required.";
+        
+        public override IconUsage? Icon => FontAwesome.Solid.DotCircle;
+        
+        public override ModType Type => ModType.Conversion;
+
+        [SettingSource("Add end notes", "Also add a note at the end of a held note")]
+        public BindableBool AddEndNotes { get; } = new BindableBool
+        {
+            Default = true,
+            Value = true
+        };
+
+        [SettingSource("Length threshold", "Only add an end note for held notes longer than this threshold (in milliseconds)")]
+        public BindableNumber<double> Threshold { get; } = new BindableDouble
+        {
+            MinValue = 1.0,
+            MaxValue = 1990.0,
+            Default = 200.0,
+            Value = 200.0,
+            Precision = 1.0,
```

Comment:
I'm not really sure this can be exposed to the user. Should be a constant value instead.

Reply:
agreed, don''t see much of a point having this as a setting either. especially with these min-max bounds (why 1ms-1990ms?)

Reply:
I did also feel a little conflicted adding it as a setting, since even I couldn't understand how the "threshold" was even supposed to work at first. If we revert it to a constant, would the 200ms be good, or some other value? 
I only added this setting out of concerns of excessive difficulty, since high-speed maps tend to get a lot of note clutter with too many end notes — hence why I thought having an option to control it might be more helpful, since it can be adapted to the difficulty of the map. I couldn't think of a good way to make this depend on how hard the map was *supposed* to be tho, so I just let the user decide. Probably overthinking this at any rate.

Reply:
A threshold/ms time value doesn't make sense since it bypasses beat snapping, so I would hope that that's used instead. In other rulesets, "taps" are typically 1/1 or 1/2 beats so I would propose the same value to be used here - that is, a hold note shorter than 1/2 beat is converted into only 1 note.

Perhaps if a setting is needed, that could be the setting. "Beat snap of notes generated from long notes" or something like that.

Reply:
I'd probably drop this setting and keep things simple to start with?

Reply:
Yeah, again I'd pick 1/1 or 1/2 beats to start off with and remove _both_ settings. 1/4 is doable (in my rusty state I can do ~140BPM 1/4s single tap), but that's already towards the upper limit for most people that I don't think will fit well in the general case.

Reply:
So should we remove the option to choose entirely? Or restrict it to more reasonable values. Forgive me for forgetting about this post, as I went and added up to 1/16 notes, which is way beyond the "doable" 1/4, so if it's like you say, I can't see this getting used by 99% of people, I suppose... shall we remove the settings menu after all?

Reply:
I'd still agree with removing the option, personally. Seems like too much customisation just for the sake of it, and it can be re-added in the future if there is actual demand for it.

Reply:
Agree with removing and waiting for there to be a demand for it. Choosing a sane setting sounds best.

My rationale is that adding settings like this come with an overhead of supporting them properly (ie `osu-web`, replays etc) and for a mod like this which is bound to see regular use, keeping the behaviour simple is in our best interest.

You're welcome to leave the functions as they are in case we need to add more flexibility in the future - just remove the enum and bindable and replace with a `const` for now.

Reply:
Thanks, this seems much more sensible. I got rid of the settings and updated the tests, so now we have it set to half-note as a constant.